File: bot.js
const { Telegraf } = require("telegraf")

const BOT_TOKEN = "8694629158:AAGaga2-MI6ZWOvcNLuVlXLWWxsWM8DqQ8A"
const ADMIN_ID = 8489998847 // твой Telegram ID

const bot = new Telegraf(BOT_TOKEN)

// старт
bot.start((ctx) => {
    ctx.reply("📩 Добро пожаловать в piski предложку\n\nОтправь сообщение, фото, видео или стикер — оно попадёт администратору.")
})

// пересылка сообщений админу
bot.on("message", async (ctx) => {

    // если сообщение от админа — пропускаем
    if (ctx.from.id == ADMIN_ID) return

    const userId = ctx.from.id
    const username = ctx.from.username || "без username"

    // пересылаем сообщение
    const msg = await ctx.forwardMessage(ADMIN_ID)

    // добавляем кнопку ответа
    await ctx.telegram.sendMessage(
        ADMIN_ID,
        `📨 Новое сообщение\n\n👤 @${username}\n🆔 ${userId}`,
        {
            reply_to_message_id: msg.message_id
        }
    )
})

// ответ админа
bot.on("reply_to_message", async (ctx) => {

    if (ctx.from.id !== ADMIN_ID) return

    const replied = ctx.message.reply_to_message

    if (!replied.forward_from) return

    const userId = replied.forward_from.id

    try {
        await ctx.copyMessage(userId)
        await ctx.reply("✅ Ответ отправлен")
    } catch {
        ctx.reply("❌ Не удалось отправить")
    }
})

bot.launch()

console.log("piski предложка бот запущен")

File: bot.py
import json
import asyncio
from aiogram import Bot, Dispatcher
from aiogram.types import Message
from aiogram.filters import Command
from aiogram.client.default import DefaultBotProperties

TOKEN = "8693534383:AAEM2Ql3WfX4iHrBOTAmBSKWNSpViYXuH2I"

bot = Bot(TOKEN, default=DefaultBotProperties(parse_mode="HTML"))
dp = Dispatcher()

POSTS_FILE = "posts.json"

posts = {}
upload_buffer = []


def load_posts():
    try:
        with open(POSTS_FILE, "r") as f:
            return json.load(f)
    except:
        return {}


def save_posts():
    with open(POSTS_FILE, "w") as f:
        json.dump(posts, f)


posts = load_posts()


# старт
@dp.message(Command("start"))
async def start(message: Message):

    args = message.text.split(" ")

    if len(args) > 1 and args[1].startswith("post_"):

        post_id = args[1].split("_")[1]

        if post_id in posts:

            for msg in posts[post_id]:
                await bot.copy_message(
                    chat_id=message.chat.id,
                    from_chat_id=msg["chat_id"],
                    message_id=msg["message_id"]
                )

        else:
            await message.answer("Файлы не найдены")

    else:

        await message.answer(
            "📁 Добро пожаловать в piski files.\n\n"
            "Этот бот позволяет получать доступ к сохранённым файлам по ссылкам."
        )


# начать загрузку
@dp.message(Command("admingetlink"))
async def admin_get(message: Message):

    global upload_buffer
    upload_buffer = []

    await message.answer(
        "📤 Ожидаю файлы.\n"
        "Отправьте один или несколько файлов."
    )


# ловим файлы
@dp.message()
async def collect_files(message: Message):

    global upload_buffer

    if message.text and message.text.startswith("/"):
        return

    upload_buffer.append({
        "chat_id": message.chat.id,
        "message_id": message.message_id
    })

    await message.answer(
        "Файл добавлен.\n"
        "Когда закончите отправку используйте /adminmakelink"
    )


# создание ссылки
@dp.message(Command("adminmakelink"))
async def make_link(message: Message):

    global upload_buffer

    if not upload_buffer:
        await message.answer("Нет файлов для создания ссылки")
        return

    post_id = str(len(posts) + 1)

    posts[post_id] = upload_buffer
    save_posts()

    bot_info = await bot.get_me()

    link = f"https://t.me/{bot_info.username}?start=post_{post_id}"

    upload_buffer = []

    await message.answer(
        f"✅ Ссылка создана:\n{link}"
    )


async def main():
    await dp.start_polling(bot)


if __name__ == "__main__":
    asyncio.run(main())